### PR TITLE
DOC-10699 PR #127186 - pkg/cli: Include both resolutions (10s and 30min) in tsdump

### PR DESCRIPTION
**Pull request by florence-crl**
Fixes DOC-10699

In cockroach-debug-tsdump.md, added section for Available resolutions.

Rendered preview:

- [Available resolutions](https://deploy-preview-19102--cockroachdb-docs.netlify.app/docs/v24.3/cockroach-debug-tsdump.html#available-resolutions)

### CODE REVIEW

File: src/current/v24.3/cockroach-debug-tsdump.md [hunk 1]
```diff
@@ -36,6 +36,18 @@ Flag | Description
...
+
+A `debug tsdump` includes all the available resolutions (10s and 30m) of the raw time series values in the time range set using the `--from` and `--to` flags:
+
+- Time series values will be dumped with a 10 seconds resolution if the supplied time range is within the 10s TTL set by the  [`timeseries.storage.resolution_10s.ttl`]({% link {{ page.version.version }}/cluster-settings.md %}#setting-timeseries-storage-resolution-10s-ttl) cluster setting (default: 10 days).
+- Time series values will be dumped with a 30 minutes resolution if the supplied time range is older than the 10s TTL back to the 30m TTL set by the [`timeseries.storage.resolution_30m.ttl`]({% link {{ page.version.version }}/cluster-settings.md %}#setting-timeseries-storage-resolution-30m-ttl) cluster setting (default: 90 days).
+
+For example, if `--from` was set to the current timestamp and `--to` was set to 15 days before the current timestamp, the `debug tsdump` would include:
+
+- For the past 1 to 10 days, time series values with a 10 seconds resolution.
+- For the past 11 to 15 days, time series values with a 30 minutes resolution.
```
Comment (arjunmahishi) [inline src/current/v24.3/cockroach-debug-tsdump.md:49]: Two suggestions here:

> if `--from` was set to the current timestamp and `--to` was set to 15 days before the current timestamp

The timestamps should be swapped. I.e `--to` should be set to current timestamp, and `--from` should be set to  `now-15days`. I think we will throw an error is `--to` < `--from`.

---

> - For the past 1 to 10 days, time series values with a 10 seconds resolution.
> - For the past 11 to 15 days, time series values with a 30 minutes resolution.

This sounds right to me. But I wonder if we can make it more clear. The **older** timestamps are downsampled to  30s. Maybe using actual dates might help:

> For example (say today was 2024-11-15) `--from` was set to `2024-11-01` and `--to` was set to `2024-11-15`, the `debug tsdump` would include:
>
> - Metrics with 10 seconds resolution for `2024-11-05` to `2024-11-15` (within the TTL).
> - And 30 minutes resolution for `2024-11-01` to `2024-11-04` (outside the T L)

Comment (florence-crl) [reply src/current/v24.3/cockroach-debug-tsdump.md]: thank you for catching the incorrect flags and the clarifying suggestion. I corrected the flags and I updated with your suggestion.

**Review by florence-crl — COMMENTED** [review]
tftr

**Review by arjunmahishi — APPROVED** [review]

File: src/current/v24.3/cockroach-debug-tsdump.md [hunk 2]
```diff
@@ -36,6 +36,18 @@ Flag | Description
...
 `--to` | The newest timestamp to include (inclusive), in the format `YYYY-MM-DD [HH:MM[:SS]]`.<br><br>**Default:** Current timestamp plus 29 hours
 
+#### Available resolutions
+
+A `debug tsdump` includes all the available resolutions (10s and 30m) of the raw time series values in the time range set using the `--from` and `--to` flags:
+
+- Time series values will be dumped with a 10 seconds resolution if the supplied time range is within the 10s TTL set by the  [`timeseries.storage.resolution_10s.ttl`]({% link {{ page.version.version }}/cluster-settings.md %}#setting-timeseries-storage-resolution-10s-ttl) cluster setting (default: 10 days).
+- Time series values will be dumped with a 30 minutes resolution if the supplied time range is older than the 10s TTL back to the 30m TTL set by the [`timeseries.storage.resolution_30m.ttl`]({% link {{ page.version.version }}/cluster-settings.md %}#setting-timeseries-storage-resolution-30m-ttl) cluster setting (default: 90 days).
+
+For example if today was `2024-11-15` and `--to` was set to the current timestamp (`2024-11-15`), and `--from` was set to 15 days before the current timestamp (`2024-11-01`), the `debug tsdump` would include:
```
Comment (kathancox) [inline src/current/v24.3/cockroach-debug-tsdump.md:46]:
```suggestion
For example, if today was `2024-11-15` and `--to` was set to the current timestamp (`2024-11-15`), and `--from` was set to 15 days before the current timestamp (`2024-11-01`), the `debug tsdump` would include:
```

**Review by florence-crl — COMMENTED** [review]
TFTRs

**Pull request merged**